### PR TITLE
Internal #2534: IGNORE NULLS Threading

### PR DESCRIPTION
**Pull request by hawkfish**
The ignore_nulls mask is just a copy of the argument mask, so we can use it directly.

### CODE REVIEW

File: src/execution/window_executor.cpp [hunk 1]
```diff
@@ -1428,22 +1428,37 @@ class WindowValueGlobalState : public WindowExecutorGlobalState {
...
 	DataChunk payload_data;
 	// The partition values
 	WindowDataChunk payload_collection;
+	// Mask to use for exclusion if we are not ignoring NULLs
+	ValidityMask no_nulls;
 	// IGNORE NULLS
-	ValidityMask ignore_nulls;
-	// Serialisation lock
-	mutex lock;
+	ValidityMask *ignore_nulls;
```
Comment (Mytherin) [inline src/execution/window_executor.cpp:1461]: nit: `optional_ptr<ValidityMask>`

Comment (Mytherin) [reply src/execution/window_executor.cpp:1461]: Actually it seems like this is always set - can it be a `reference<ValidityMask>` instead?

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM - one minor comment

File: src/execution/window_executor.cpp (hunk 1 above)
Comment (hawkfish) [reply src/execution/window_executor.cpp:1461]: That would require moving code around. And my apologies for always forgetting about smart pointers!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**